clang/lib/AST/Interp/ByteCodeStmtGen.cpp

```diff
@@ -628,6 +630,12 @@ bool ByteCodeStmtGen<Emitter>::visitAsmStmt(const AsmStmt *S) {
   return this->emitInvalid(S);
 }
 
+template <class Emitter>
+bool ByteCodeStmtGen<Emitter>::visitAttributedStmt(const AttributedStmt *S) {
+  // Ignore all attributes.
```

Comment:
Is this what we want to do for `[[assume]]`? We can do it this way: https://eel.is/c++draft/expr.const#5.8 but we can also do it better: https://eel.is/c++draft/expr.const#5.33

CC @erichkeane @cor3ntin

(We don't support assumptions yet, so I think the code is fine as-is, but I'm wondering if we want to add a FIXME here about `[[assume]]`. We should also see if any of our vendor statement attributes have UB we need to consider as well.)

Reply:
We can add a fix me but i wouldn't worry about it too much, until we have a clearer idea of what we do, which we really don't atm
And the wording is clear that this is unspecified either way.

And for vendor attributes, it would only make sense to check them if we want them to fail the evaluation, do we have such things?

Reply:
> Is this what we want to do for [[assume]]? We can do it this way: https://eel.is/c++draft/expr.const#5.8 but we can also do it better: https://eel.is/c++draft/expr.const#5.33

The second case would essentially work like an assertion? If this fails, the statement isn't a constant expression?

Reply:
At compile time, i suppose so!

Reply:
I'm sitting in core, which is a good time to mention https://wg21.link/cwg2763 - `[[assume]]` is not the only UB standard attribute - but it requires a specific handling so it is also orthogonal to this patch